### PR TITLE
Feature/multiple choice test

### PR DESCRIPTION
**Pull request by JTrillo**
This PR solves issue #65.
@iestevea and me have developed this new test. Now, we are going to describe the changes made:

- Test buttons in home component are sorted according to test difficulty. Thus, **multiple choice** test is the first one, **fill the gap** the second, and **irregular verbs** is the last one.
- The test itself: scene and pod.
- A new route entry in the router (for the new test).
- A requirement in verbs settings. Since this test needs 3 different verbs (the correct one and 2 other options), we have made a check in verb setting to verify that at least 3 verbs are selected. This minimum number can be modified in file **src/core/const.ts**.
- Minor styles modifications.

A couple of screenshots will be posted on next comments

### CODE REVIEW

**Comment by JTrillo** [comment]
image

**Comment by JTrillo** [comment]
image

**Comment by JTrillo** [comment]
image

File: src/pods/test-multiple-choice/test-multiple-choice.business.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { VerbEntityGlobal } from 'core/verbs';
+import { VerbQuiz } from './test-multiple-choice.vm';
+import { Verb, VerbCorrect, createDefaultVerbCorrect } from 'common/model';
+
+// TODO: maybe add some defensive programming here? edge cases / errors ?
```
Comment (brauliodiez) [inline src/pods/test-multiple-choice/test-multiple-choice.business.ts:5]: This two TODO we should decide whether to remove them or do something about them

File: src/pods/test-multiple-choice/test-multiple-choice.business.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  let verbCorrect = createDefaultVerbCorrect();
+  verbCorrect.all = verb.infinitive === quiz.response;
+  return verbCorrect;
+};
+
+export const mixOptions = (options: Verb[]) => {
+  for (let i = options.length - 1; i > 0; i--) {
+    let j = Math.floor(Math.random() * (i + 1));
+    [options[i], options[j]] = [options[j], options[i]];
+  }
```
Comment (brauliodiez) [inline src/pods/test-multiple-choice/test-multiple-choice.business.ts:34]: This is a little bit criptic, what do we want to achieve here? My guess

- I start from the last option to the first one.
- I just pick a random number in between the current index plus one and cero.
-I swap values (mutating the array).

Before refactoring we should create some unit tests for this cases:
  - We cannot test that numbers are swapped because it could happen that it doesn't change.
  - We can try with arrays of several length.
  - We can test that array length is the same.
  - We can test that every verb on the original array are in the resulting one.

Trying to apply some clean code, first let's go for a mutation version (not ideal), then let's try an inmmutable version:

```ts
export const mixOptions = (options: Verb[]) => {
  const mixedOptions = [...options];

  for (let currentPosition = mixedOptions.length - 1; currentPosition > 0; currentPosition--) {
    let newPosition = Math.floor(Math.random() * (currentPosition + 1));
    [options[currentPosition], options[newPosition]] = [options[newPosition], options[currentPosition]];
  }

  return mixedOptions;
}
```

File: src/pods/test-multiple-choice/test-multiple-choice.container.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        otherRandom = pickRandomVerb(selectedVerbs, verbCollection);
+      } while(randomVerb.infinitive === otherRandom.infinitive);
+      setOtherOption1(otherRandom);
+      // Check the second other option is not repeated
+      do {
+        otherRandom2 = pickRandomVerb(selectedVerbs, verbCollection);
+      } while(randomVerb.infinitive === otherRandom2.infinitive ||
+              otherRandom.infinitive === otherRandom2.infinitive);
+      setOtherOption2(otherRandom2);
+    }
```
Comment (brauliodiez) [inline src/pods/test-multiple-choice/test-multiple-choice.container.tsx:57]: This code can be extracted to a _test-multiple-choice.business.tsx_ it could be something like:

_test-multiple-choice.container.tsx_
```tsx
React.useEffect(() => {
  const randomVerb = pickRandomVerb(selectedVerbs, verbCollection);
  setCurrentVerb(randomVerb);
  const otherVerbOptions = pickFalseVerbOptions(rightVerbOption);
  setOtherOption1(otherVerbOptions[1]);
  setOtherOption2(otherVerbOptions[2]);
},[currentQuestion]
```

Pseudocode, add unit tests to this

_test-multiple-choice.business.ts_
```ts
const pickFalseVerbOptions = (rightVerbOption : string) => {
   const optionA = [pickOption(rightVerbOption)];
   const optionB = [optionA, pickOption(rigthVerbOption)];

  return [optionA, optionB];
}

const pickOption = (optionsExcluded : string[]) : string => {
      do {
        selectedOption = pickRandomVerb(selectedVerbs, verbCollection);
      } while(optionsExcluded.findIndex(verb => verb === selectedOption.infinitive) != -1);

    return selectedOption;
}
```

Then we can add unit tests to the business part

**Comment by JTrillo** [comment]
It's not ready yet. I need to add specs to each test business

File: src/pods/test-multiple-choice/test-multiple-choice.business.ts (hunk 1 above)
Comment (JTrillo) [reply src/pods/test-multiple-choice/test-multiple-choice.business.ts]: I've removed both TODO's and moved **pickRandomVerb** function to common

File: src/pods/test-multiple-choice/test-multiple-choice.business.ts (hunk 2 above)
Comment (JTrillo) [reply src/pods/test-multiple-choice/test-multiple-choice.business.ts]: I've installed and used **lodash.shuffle**

File: src/pods/test-multiple-choice/test-multiple-choice.container.tsx (hunk 3 above)
Comment (JTrillo) [reply src/pods/test-multiple-choice/test-multiple-choice.container.tsx:49]: I've moved the function to business file. I've also created a spec file on each pod in order to test business files
